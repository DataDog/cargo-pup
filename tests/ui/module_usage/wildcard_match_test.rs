//@compile-flags: --crate-name wildcard_match
//@compile-flags: --crate-type lib

// Test demonstrating wildcard matches in import paths

// Module with imports that should only allow certain modules
mod test_wildcard_match {

    // This import is denied by the regex pattern
    use std::collections::HashMap; //~ ERROR: Use of module 'std::collections::HashMap' is denied; ["^std::.*"] are not permitted.

    // This import is also denied by the regex pattern
    use std::env; //~ ERROR: Use of module 'std::env' is denied; ["^std::.*"] are not permitted.

    fn test_function() {
        let mut map = HashMap::new();
        map.insert("key", "value");
        let _ = env::var("HOME");
    }
}
