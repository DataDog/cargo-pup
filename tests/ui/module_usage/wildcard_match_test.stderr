error: Use of module 'std::collections::HashMap' is denied; ["^std::.*"] are not permitted.
  --> tests/ui/module_usage/wildcard_match_test.rs:10:5
   |
LL |     use std::collections::HashMap;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Remove this import.
   = note: Applied by cargo-pup rule 'wildcard_match_rules'.
   = note: `#[deny(module_usage)]` on by default

error: Use of module 'std::env' is denied; ["^std::.*"] are not permitted.
  --> tests/ui/module_usage/wildcard_match_test.rs:13:5
   |
LL |     use std::env;
   |     ^^^^^^^^^^^^^
   |
   = help: Remove this import.
   = note: Applied by cargo-pup rule 'wildcard_match_rules'.

error: aborting due to 2 previous errors

